mule-classloader-model/src/main/java/org/mule/tools/api/classloader/model/LegacyApplicationClassloaderModel.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Mule ESB Maven Tools
+ * <p>
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * <p>
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tools.api.classloader.model;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+
+import java.util.Collection;
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+
+public class LegacyApplicationClassloaderModel extends ApplicationClassloaderModel {
+
+  public LegacyApplicationClassloaderModel(ClassLoaderModel classLoaderModel) {
+    super(classLoaderModel);
+  }
+
+  @Override
+  public void mergeDependencies(Collection<ClassLoaderModel> otherClassloaderModels) {
+    for (ClassLoaderModel otherClassLoaderModel : otherClassloaderModels) {
```

Comment:
isn't this method really similar to the one implemented in the parent class? Maybe use a template method to avoid duplicating code

Reply:
Ok, I can have a utility method with a callback